pkg/kubectl/cmd/create_podsecuritypolicy.go

```diff
@@ -0,0 +1,171 @@
+	Mapper       meta.RESTMapper
+	Out          io.Writer
+	PrintObject  func(obj runtime.Object) error
+}
+
+func NewCmdCreatePSP(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	c := &CreatePSPOptions{
+		Out: cmdOut,
+	}
+	cmd := &cobra.Command{
+		Use: "podsecuritypolicy NAME --selinux=rule --supplemental-groups=rule --run-as-user=rule --fs-group=rule [other options] [--dry-run]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"psp"},
+		Short:                 PSPLong,
+		Long:                  PSPLong,
+		Example:               pspExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(c.Complete(f, cmd, args))
+			cmdutil.CheckErr(c.Validate())
+			cmdutil.CheckErr(c.RunCreatePSP())
+		},
+	}
+
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddPrinterFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+	cmd.Flags().String("selinux", "RunAsAny", "define the strategy that will dictate the allowable labels that may be set")
+	cmd.Flags().String("run-as-user", "RunAsAny", "control what user ID containers can run as.")
+	cmd.Flags().String("supplemental-groups", "RunAsAny", "control which group IDs containers can add.")
+	cmd.Flags().String("fs-group", "RunAsAny", "provide a whitelist of allowed volume types.")
+
+	return cmd
+}
+
+func (c *CreatePSPOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+
+	name, err := NameFromCommandArgs(cmd, args)
+	if err != nil {
+		return err
+	}
+	c.Name = name
+	c.SELinux = cmdutil.GetFlagString(cmd, "selinux")
+	c.RunAsUser = cmdutil.GetFlagString(cmd, "run-as-user")
+	c.SupplementalGroups = cmdutil.GetFlagString(cmd, "supplemental-groups")
+	c.FSGroup = cmdutil.GetFlagString(cmd, "fs-group")
+	// Complete other options for Run.
+	c.Mapper, _ = f.Object()
+
+	c.DryRun = cmdutil.GetDryRunFlag(cmd)
+	c.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+
+	c.PrintObject = func(obj runtime.Object) error {
+		return cmdutil.PrintObject(cmd, obj, c.Out)
+	}
+	clientset, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+	c.Client = clientset.ExtensionsV1beta1()
+
+	return nil
+}
+
+func (c *CreatePSPOptions) Validate() error {
+	if c.Name == "" {
+		return fmt.Errorf("name must be specified")
+	}
+	if c.SELinux != "RunAsAny" && c.SELinux != "MustRunAs" {
+		return fmt.Errorf("illegal --selinux rule, must be one of RunAsAny MustRunAs")
+	}
+	if c.RunAsUser != "RunAsAny" && c.RunAsUser != "MustRunAs" && c.RunAsUser != "MustRunAsNonRoot" {
+		return fmt.Errorf("illegal --run-as-user rule, must be one of RunAsAny MustRunAs MustRunAsNonRoot")
+	}
+	if c.SupplementalGroups != "RunAsAny" && c.SupplementalGroups != "MustRunAs" {
+		return fmt.Errorf("illegal --supplemental-groups rule, must be one of RunAsAny MustRunAs")
+	}
+	if c.FSGroup != "RunAsAny" && c.FSGroup != "MustRunAs" {
+		return fmt.Errorf("illegal --fs-group rule, must be one of RunAsAny MustRunAs")
+	}
```

Comment:
I'd like to re-use validation code from `pkg/apis/extensions/validation/validation.go` if it's possible.

Anyway, please, use constants instead of hard-coded strings (`RunAsAny` -> `extensions.SupplementalGroupsStrategyRunAsAny` and so on).